### PR TITLE
Return values and error checking

### PR DESCRIPTION
**Pull request by gmegan**
Resolve issues #25 and #27

### CODE REVIEW

File: content/shmem_team_destroy.tex [hunk 1]
```diff
@@ -20,16 +20,20 @@
 handle can no longer be used for team API calls.
 
 It is considered erroneous to free \LibHandleRef{SHMEM\_TEAM\_WORLD} or
-any other predefined team. Error checking will be done to ensure a valid
-team handle is provided. Errors will result in a return value less than \CONST{0}.
+any other predefined team.
+
+If a pointer to an invalid handle is provided, behavior is undefined.
```
Comment (nspark) [inline content/shmem_team_destroy.tex]: Suggest: "If a pointer to an invalid handle is provided, **the** behavior is undefined."

(occurs more times below)

File: content/shmem_team_destroy.tex [hunk 2]
```diff
@@ -20,16 +20,20 @@
...
+
+If the pointer to \VAR{team} is a null pointer, no team is destroyed,
```
Comment (nspark) [inline content/shmem_team_destroy.tex]: Suggest: "If the pointer to `team` is a null pointer, **then** no team is destroyed, ..."

File: content/shmem_team_split_2d.tex [hunk 3]
```diff
@@ -96,17 +96,19 @@
...
 }
 
-If \VAR{parent\_team} is equal to \LibConstRef{SHMEM\_TEAM\_NULL}, both
-\VAR{xaxis\_team} and \VAR{yaxis\_team} will be assigned the value
-\LibConstRef{SHMEM\_TEAM\_NULL}.
-Otherwise, if \VAR{parent\_team} is an invalid team handle,
-the behavior is undefined.
+If \VAR{parent\_team} is an invalid team handle, the behavior is undefined.
+
+If \VAR{parent\_team} is equal to \LibConstRef{SHMEM\_TEAM\_NULL}, no new
```
Comment (nspark) [inline content/shmem_team_split_2d.tex]: Suggest: "is equal to" → "compares equal to"

(repeats more below)

**Comment by gmegan** [comment]
Updated phrasing as suggested. Also replaced "function" with "routine" in these files.

**Pull request merged**